framework/src/main/java/org/checkerframework/framework/flow/CFAbstractValue.java

```diff
@@ -277,7 +283,46 @@ public V mostSpecific(@Nullable V other, @Nullable V backup) {
     if (ms.error) {
       return backup;
     }
-    return analysis.createAbstractValue(mostSpecific, mostSpecifTypeMirror);
+    V mostSpecificVal = analysis.createAbstractValue(mostSpecific, mostSpecifTypeMirror);
+    if (this.getThenStore() != null) {
+      mostSpecificVal.setStores(thenStore, elseStore);
```

Comment:
Why is it OK to ignore `other`'s stores if `this` has a store?  Might it have information that should be retained?